### ISSUE TITLE
Consider accept-language header when rendering `<App>` server-side

### ISSUE THREAD

**Issue by eric-burel** [VulcanJS/Vulcan#2055]
## Issue:
Out-of-the-box SSR will ignore the request `accept-language` header, so it fallback to default language all the time. Hardcoding is a solution but it require changes in the user preferred locale and `cookies` does not seem to work yet.
Note that it works client side as expected since the locale is available. So SSR will trigger a warning, and the site will be displayed in english (or not displayed if en_EN token are not set) before being translated to the correct locale.

## Proposed fix:
- Implement a provider that takes pass the `accept-language` first language to its children
```js
// Ugly demo that might be slightly wrong but does the job
import React from 'react'

const RequestLocaleProvider = ({ req, children }) => {
    const locale = req.headers["accept-language"].split(',')[0].split('-')[0]
    return React.cloneElement(React.Children.only(children), { requestLocale: locale })
}
export default RequestLocaleProvider
```

- pass it to the `app` server side. This I did not figure out. I tried to change the `server/routing.jsx ` file of `vulcan:router`.

Basically I added my wrapper to the `{app}`, but the `requestLocale` prop is still not available during SSR.

```js
    wrapperHook(req, res, appGenerator) {
      const { apolloClient, store } = getRenderContext();
      store.reload();
      store.dispatch({ type: '@@nova/INIT' }) // the first dispatch will generate a newDispatch function from middleware
      const app = runCallbacks('router.server.wrapper', appGenerator(), { req, res, store, apolloClient });
      return <ApolloProvider store={store} client={apolloClient}><CookiesProvider cookies= 
      {req.universalCookies}><RequestLocaleProvider req={req}>{app} 
       </RequestLocaleProvider></CookiesProvider></ApolloProvider>
    }
```

**Comment by SachaG** [comment]
I just fixed this. I did something similar but the tough part was passing the correct header to the GraphQL endpoint.

**Comment by eric-burel** [comment]
Awesome! I'll test that, thanks for the huge work on this.
